TESTING.md

```diff
@@ -0,0 +1,274 @@
+components. For example, if documentation says that a component will pass extra props to a container
+component, there should be a specification that verifies this claim. If a component's documentation
+states that children will be rendered, a test should verify the children exist within the containing
+component.
+
+Good unit tests avoid too much implementation detail. For example, if a component states it will
+render the children prop, the documentation and specifications shouldn't state where unless it is
+specific to the functionality of the component. Tests here should define the API surface area that
+will be guaranteed. The more implementation details, the more difficult the tests are to maintain.
+
+**Bad**:
+
+```ts
+const {container} = render(
+  <SomeComponent>
+    <div data-testid="test">Test</div>
+  </SomeComponent>
+);
+
+expect(container.children[0].children[0].textContent).toEqual('Test');
+```
+
+This test encodes the DOM structure into it. Any changes to the source code will require a change to
+the test making the test tightly coupled and fragile.
+
+**Good**:
+
+```ts
+const {getByTestId} = render(
+  <SomeComponent data-testid="container">
+    <div data-testid="test">Test</div>
+  </SomeComponent>
+);
+
+const component = await getByTestId('container');
+const child = await getByTestId('test');
+
+expect(component).toContainElement(child);
+```
+
+The first test has the DOM structure encoded in it. The second
+
+This example is more flexible. The component can be refactored with DOM elements added or removed
+and the test will still pass as long as the `div` is a child of the component somewhere (the
+specification is met). Also this example will have a more useful failure message since
+`.toContainElement` has the context that it is expecting an element in another element vs a match of
+a string.
+
+## Functional tests
+
+Canvas Kit uses [Cypress][https://cypress.io] for browser-based behavior testing (additional info:
+[Why Cypress?](./cypress/WHY_CYPRESS.md)). Functional tests ensure the code meets functional
+specifications from a user's perspective. Specifications can come from many different stakeholders
+including product management, user research and accessibility. A functional test should read in the
+Given/When/Then format and fits in well as acceptance criteria.
+
+Example of a usability specification: "Given the modal example, when a user clicks the 'Delete Item'
+button, then the modal opens".
+
+Example of an accessibility specification: "Given the modal example, when a user clicks the 'Delete
+Item' button, then the focus should be transferred to the first focusable element in the dialog"
+
+The BDD-style (behavior-driven development) syntax lends well to the nesting of actions performed on
+a example (also called "story" borrowed from Storybook nomenclature). The wording used inside the
+BDD blocks should read in English and be understandable by non-developers. The implementation should
```

Comment:
It sounds like you're specifying the English language rather than what I think you're trying to say:

```suggestion
BDD blocks should read in plain English and be understandable by non-developers. The implementation should
```